modules/serverpod_auth/serverpod_auth_server/lib/src/business/user_authentication.dart

```diff
@@ -4,20 +4,31 @@ import 'package:serverpod_auth_server/module.dart';
 import 'package:serverpod_auth_server/src/business/authentication_util.dart';
 import 'package:serverpod_shared/serverpod_shared.dart';
 
-/// Collects methods for authenticating users.
+/// Provides methods for authenticating users.
 class UserAuthentication {
-  /// Signs in an user to the server. The user should have been authenticated
-  /// before signing them in. Send the AuthKey.id and key to the client and
-  /// use that to authenticate in future calls. In most situations you should
-  /// use one of the auth providers instead of this method.
+  /// Signs in a user and generates an authentication key.
+  /// Sends the `AuthKey.id` and `key` to the client for future authentication.
+  ///
+  /// Before calling this method, ensure the user has been authenticated
+  /// through appropriate methods (e.g., email, social sign-ins).
+  /// This method only generates the authentication key after successful
+  /// authentication.
+  ///
+  /// The authenticated user will be signed into the provided session,
+  /// and their session will be updated with the user's authentication info.
```

Comment:
Suggestion: This is only true if the `updateSession` bool is true.

Please harmonize this.